src/aws.ts

```diff
@@ -504,6 +507,237 @@ export namespace $AWS {
+                      })
+                    );
+                  },
+                  preWarm(preWarmContext) {
+                    preWarmContext.getOrInit({
+                      key: `$AWS.SDK.${serviceName}`,
+                      init: (key, props) =>
+                        new AWS[serviceName](
+                          props?.clientConfigRetriever?.(key)
+                        ),
+                    });
+                  },
+                  call(args, preWarmContext) {
+                    const client: any = preWarmContext.getOrInit({
+                      key: `$AWS.SDK.${serviceName}`,
+                      init: (key, props) =>
+                        new AWS[serviceName](
+                          props?.clientConfigRetriever?.(key)
+                        ),
+                    });
+
+                    return client[methodName](args[0]).promise();
+                  },
+                },
+                asl: (call, context) => {
+                  const iamOptions = call.args[1] as unknown as IamOptions;
+
+                  if (!isObjectLiteralExpr(iamOptions)) {
+                    throw new SynthError(
+                      ErrorCodes.Expected_an_object_literal,
+                      `Second argument ('iamOptions') into a SDK call must be an object`
+                    );
+                  }
+
+                  context.role.addToPrincipalPolicy(
+                    new aws_iam.PolicyStatement({
+                      effect: aws_iam.Effect.ALLOW,
+                      actions: iamOptions.iamActions ?? defaultIamActions,
+                      resources: iamOptions.iamResources,
+                      conditions: iamOptions.iamConditions,
+                    })
+                  );
+
+                  const sdkIntegrationServiceName =
+                    mapAslSdkServiceName(serviceName);
+                  const input = call.args[0]?.expr;
+
+                  if (!input) {
+                    throw (
+                      (new SynthError(ErrorCodes.Invalid_Input),
+                      "SDK integrations need a single input")
+                    );
+                  }
+
+                  return context.evalExpr(input, (output) => {
+                    if (
+                      !ASLGraph.isLiteralValue(output) ||
+                      typeof output.value !== "object" ||
+                      !output.value
+                    ) {
+                      throw new SynthError(
+                        ErrorCodes.Unexpected_Error,
+                        "Expected an object literal as the first parameter."
+                      );
+                    }
+
+                    return context.stateWithHeapOutput({
+                      Type: "Task",
+                      Resource: `arn:aws:states:::aws-sdk:${sdkIntegrationServiceName}:${methodName}`,
+                      Parameters: output.value,
+                      Next: ASLGraph.DeferNext,
+                    });
```

Comment:
if you want to support more than just a LiteralValue, you can do

```ts
// normalized any output to a jsonPath or literal
return context.evalExprToJsonPathOrLiteral(input, (output) => {
    return context.stateWithHeapOutput(
        // can add LiteralValue or JsonPath as the parameter to a task.
        ASLGraph.taskWithInput({
            Type: "Task",
            Resource: `arn:aws:states:::aws-sdk:${sdkIntegrationServiceName}:${methodName}`,
            Next: ASLGraph.DeferNext,
       }, output);
   });
);
```

Reply:
Unfortunately I get `Property 'evalExprToJsonPathOrLiteral' does not exist on type 'ASL'.ts(2339)` ... what is the correct method here?

Reply:
Woops, forgot to mention they were new, haha. That change added full support for conditions anywhere in the source.